tests/unit/invert/test_fft.cxx

```diff
@@ -0,0 +1,149 @@
+#include "gtest/gtest.h"
+
+#include "bout/array.hxx"
+#include "bout/constants.hxx"
+#include "dcomplex.hxx"
+#include "fft.hxx"
+#include "test_extras.hxx"
+
+#include <algorithm>
+#include <iostream>
+#include <numeric>
+
+TEST(FFTTest, rfft) {
+
+  // Make sure fft functions are quiet by setting fft_measure to false
+  bout::fft::fft_init(false);
+
+  constexpr int size{8};
+
+  // Calculate sin(x) + cos(2x) on [0, 2pi]
+  Array<BoutReal> indices(size);
+  std::iota(indices.begin(), indices.end(), 0);
```

Comment:
Should this really be `0.0` so that `std::iota` generates reals rather than ints that get converted?

Reply:
Also a comment to note that after this call `indices` contains  {0,1,2...,size-1} might help.